@xen-orchestra/defined/index.js

```diff
@@ -14,15 +14,11 @@
 // ])
 // ```
 function defined() {
-  let args = arguments
-  let n = args.length
-  if (n === 1) {
-    args = arguments[0]
-    n = args.length
-  }
+  const args = [...arguments].flat()
+  const n = args.length
```

Comment:
Do not change implementation unless necessary, which I don't think is the case here.

Reply:
This is to handle arrays as arguments, as specified in the Usage.md.

Reply:
I disagree, it was already working, the only problem was the line 25.

Reply:
Previously this worked with only one table if multiple were provided.

defined([undefined, undefined], [undefined, undefined, 10])
before:
// Returns [undefined, undefined]
after:
// Returns 10

Reply:
The first behavior is the correct one, the first argument is not `undefined`, it is an array.

Reply:
So we should change (again) the other test.

If we return the array on this:
  it('should return first non undefined value in arrays', () => {
    const expected = [undefined, undefined, undefined]
    assert.deepStrictEqual(defined([undefined, undefined, undefined], [undefined, undefined, 10]), expected)
  })

We should return the array on this one too, the first argument is not undefined, it is an array:

  it('should return first non undefined value in array', () => {
    const expected = null
    assert.deepStrictEqual(defined([undefined, null, 10]), expected)
  })

Reply:
Use code blocks with syntax highlighting for your future messages please :slightly_smiling_face: 

And no, you're wrong, there is two way to use this function:

- either with a single array argument: it should return the first non-`undefined` item
- or with multiple arguments: it should return the first non-`undefined` argument